File: source/livana-be/src/main/java/octguy/livanabe/dto/request/CreatePaymentRequest.java
package octguy.livanabe.dto.request;

import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import octguy.livanabe.enums.BookingType;

import java.util.UUID;

@Getter
@Setter
public class CreatePaymentRequest {
    
    @NotNull(message = "Booking ID is required")
    private UUID bookingId;
    
    @NotNull(message = "Booking type is required")
    private BookingType bookingType;
    
    private String bankCode;
    
    private String language = "vn";
}

File: source/livana-fe/src/types/request/createPaymentRequest.ts
export const BookingType = {
  HOME: "HOME",
  EXPERIENCE: "EXPERIENCE",
} as const;

export type BookingType = (typeof BookingType)[keyof typeof BookingType];

export interface CreatePaymentRequest {
  bookingId: string;
  bookingType: BookingType;
  amount: number;
  orderInfo?: string;
  bankCode?: string;
  language?: string;
}
